provisioning/transport/http/src/http.ts

```diff
@@ -0,0 +1,251 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+import { EventEmitter } from 'events';
+import { RestApiClient, Http as Base } from 'azure-iot-http-base';
+import { errors, X509 } from 'azure-iot-common';
+import * as machina from 'machina';
+
+export type ProvisioningResponseCallback = (err: Error, result?: any) => void;
+
+const _defaultHeaders = {
+  'Accept' : 'application/json',
+  'Content-Type' : 'application/json; charset=utf-8'
+};
+
+export class Http extends EventEmitter {
+
+  private _idScope: string;
+  private _registrationId: string;
+  private _body: any;
+  private _forceRegistration: boolean;
+  private _operationId: string;
+
+  private _fsm: machina.Fsm;
+  private _apiVersion: string = '2017-08-31-preview';
+  private _provisioningServer: string = 'global.azure-devices-provisioning.net';
+  private _httpTimeout: number = 4000;
+  private _operationStatusPollingInterval: number = 2000;
+  private _pollingTimer: any;
+  private _restApiClient: RestApiClient;
+  private _registrationCallback: ProvisioningResponseCallback;
+
+  /**
+   * @private
+   * @constructor
+   * @param config The configuration object.
+   */
+  /* Codes_SRS_NODE_PROVISIONING_HTTP_18_001: [ The `Http` constructor shall accept the following properties:
+  - `idScope` - a string specifiying the scope of the provisioning operations,
+  - `registrationId` - the registration id for the specific device ] */
+  constructor(idScope: string, registrationId: string, httpBase?: Base) {
+    super();
+    this._idScope = idScope;
+    this._registrationId = registrationId;
+
+    this._fsm = new machina.Fsm({
+      namespace: 'http-provisioning',
+      initialState: 'idle',
+      states: {
+        idle: {
+          _onEnter: (callback, err, response) => {
+            this._pollingTimer = null;
+            this._restApiClient = null;
+            this._body = null;
+            this._forceRegistration = false;
+            this._operationId = '';
+            this._registrationCallback = null;
+            if (callback) {
+              callback(err, response);
```

Comment:
>response [](start = 28, length = 8)

nit: response is usually used for the HTTP response object and result for the actual result (usually the JSON.parsed body of the response)